ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/behaviorTree/RDXBehaviorTreeNodeBuilder.java

```diff
@@ -2,182 +2,107 @@
+import us.ihmc.behaviors.behaviorTree.condition.*;
+import us.ihmc.behaviors.behaviorTree.control.*;
+import us.ihmc.behaviors.behaviorTree.control.ai2r.*;
+import us.ihmc.behaviors.behaviorTree.control.buildingExploration.*;
+import us.ihmc.behaviors.behaviorTree.control.door.*;
 import us.ihmc.communication.crdt.CRDTInfo;
 import us.ihmc.rdx.behaviorTree.actions.*;
+import us.ihmc.rdx.behaviorTree.condition.*;
+import us.ihmc.rdx.behaviorTree.control.*;
 import us.ihmc.rdx.ui.RDX3DPanel;
 import us.ihmc.rdx.ui.RDXBaseUI;
-import us.ihmc.rdx.behaviorTree.control.RDXAI2RNode;
-import us.ihmc.rdx.behaviorTree.control.RDXDoorTraversal;
-import us.ihmc.rdx.behaviorTree.control.RDXBuildingExploration;
-import us.ihmc.rdx.behaviorTree.condition.RDXConditionNode;
-import us.ihmc.rdx.behaviorTree.control.RDXGotoNode;
-import us.ihmc.rdx.behaviorTree.actions.RDXActionNode;
-import us.ihmc.rdx.behaviorTree.control.RDXActionSequence;
-import us.ihmc.rdx.behaviorTree.control.RDXFallbackNode;
 import us.ihmc.robotics.physics.RobotCollisionModel;
 import us.ihmc.robotics.referenceFrames.ReferenceFrameLibrary;
 import us.ihmc.robotics.robotSide.RobotSide;
 import us.ihmc.tools.io.WorkspaceResourceDirectory;
 
 import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
 
 public class RDXBehaviorTreeNodeBuilder implements BehaviorTreeNodeBuilder<RDXBehaviorTreeNode<?, ?>>
 {
+   private static final Map<Class<?>, BiFunction<Long, RDXBehaviorTreeRootNode, RDXBehaviorTreeNode<?, ?>>> MAP = new HashMap<>();
+   static
+   {
+      MAP.put(BehaviorTreeNodeDefinition.class, RDXBehaviorTreeNode::new);
+      MAP.put(AI2RNodeDefinition.class, RDXAI2RNode::new);
+      MAP.put(ActionSequenceDefinition.class, RDXActionSequence::new);
+      MAP.put(FallbackNodeDefinition.class, RDXFallbackNode::new);
+      MAP.put(ConditionNodeDefinition.class, RDXConditionNode::new);
+      MAP.put(GotoNodeDefinition.class, RDXGotoNode::new);
+      MAP.put(CheckPointNodeDefinition.class, RDXCheckPointNode::new);
+      MAP.put(DoorTraversalDefinition.class, RDXDoorTraversal::new);
+      MAP.put(BuildingExplorationDefinition.class, RDXBuildingExploration::new);
+      MAP.put(ChestOrientationActionDefinition.class, RDXChestOrientationAction::new);
+      MAP.put(FootstepPlanActionDefinition.class, RDXFootstepPlanAction::new);
+      MAP.put(HandPoseActionDefinition.class, RDXHandPoseAction::new);
+      MAP.put(HandWrenchActionDefinition.class, RDXHandWrenchAction::new);
+      MAP.put(ScrewPrimitiveActionDefinition.class, RDXScrewPrimitiveAction::new);
+      MAP.put(PelvisHeightOrientationActionDefinition.class, RDXPelvisHeightOrientationAction::new);
+      MAP.put(SakeHandCommandActionDefinition.class, RDXSakeHandCommandAction::new);
+      MAP.put(WaitDurationActionDefinition.class, RDXWaitDurationAction::new);
+      MAP.put(FootPoseActionDefinition.class, RDXFootPoseAction::new);
```

Comment:
Node builders have been simplified. They'll no longer need to change to access different things.